### PR TITLE
Admission controller

### PR DESCRIPTION
**Pull request by cezarybednarz**

### CODE REVIEW

**Comment by zimnx** [comment]
Why new PR was opened? This one lacks history of comments, i can't check whether you addressed them or how.

**Comment by cezarybednarz** [comment]
Link to previous pull request with history of comments: https://github.com/scylladb/scylla-operator-autoscaler/pull/17

File: cmd/admission-controller/admission_controller.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			webhookServer := mgr.GetWebhookServer()
+
+			logger.Info(ctx, "registering webhooks to the webhook server")
+			webhookServer.Register("/validate-scylla-scylladb-com-v1-scyllacluster", &webhook.Admission{
+				Handler: &admission_controller.AdmissionValidator{
+					Client:                        mgr.GetClient(),
+					Logger:                        logger,
+					ScyllaClient:                  client,
+					UpdaterServiceAccountUsername: updaterServiceAccountUsername,
+					ScaledResources:               []string{"cpu"},
```
Comment (zimnx) [inline cmd/admission-controller/admission_controller.go]: accept as app parameter

File: pkg/admission_controller/validatingwebhook_test.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+			scas:            autoModeDoubleScaList,
+			scaledResources: []string{"cpu"},
+			allowed:         false,
+		},
+		{
+			name:            "allow changing CPU resources when only Memory is scaled",
+			cluster:         doubleRackWithChangedCPU,
+			oldCluster:      doubleRackCluster,
+			scas:            autoModeDoubleScaList,
+			scaledResources: []string{"Memory"},
```
Comment (zimnx) [inline pkg/admission_controller/validatingwebhook_test.go]: `Memory` -> `memory`

**Review by zimnx — CHANGES_REQUESTED** [review]
small nits

File: config/default/admission_controller_webhook_patch.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  name: admission-controller
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+        - name: admission-controller
+          ports:
+            - containerPort: 443
```
Comment (zimnx) [inline config/default/admission_controller_webhook_patch.yaml]: 9443

**Review by zimnx — COMMENTED** [review]
rebase is needed and one comment

File: config/webhook/service.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: webhook-service
+  namespace: system
+spec:
+  ports:
+    - port: 443
+      targetPort: webhook
```
Comment (zimnx) [inline config/webhook/service.yaml]: `webhook-server`

**Review by zimnx — APPROVED** [review]

**Pull request merged**